rholang/src/test/scala/coop/rchain/rholang/ResourceTools.scala

```diff
@@ -0,0 +1,46 @@
+package coop.rchain.rholang
+import java.io.File
+import java.nio.file.{Files, Path}
+
+import cats.{FlatMap, MonadError}
+import cats.implicits._
+import com.typesafe.scalalogging.Logger
+
+import scala.reflect.io.Directory
+
+object ResourceTools {
+  def withResource[F[_]: FlatMap, T, R](
+      acquire: => F[R],
+      release: (R, Option[Throwable]) => F[Unit]
+  )(job: R => F[T])(implicit F: MonadError[F, Throwable]): F[T] =
+    for {
+      resource <- acquire
+      attempt  <- job(resource).attempt
+      result <- attempt.fold(
+                 ex => release(resource, Some(ex)) *> F.raiseError(ex),
+                 r => release(resource, None) *> F.pure(r)
+               )
+    } yield result
```

Comment:
Aren't you implementing the cats' `Resource` typeclass by hand?

https://typelevel.org/cats-effect/datatypes/resource.html#with-javaio-using-io

Reply:
Yes, I am. Initially I wasn't aware that cats.Resource does exactly what I needed. In the end I refactored everything to use cats.Resource. See last commits